engine/src/main/java/org/hibernate/search/engine/cfg/spi/ParseUtils.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.cfg.spi;
+
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+import static java.time.temporal.ChronoField.YEAR;
+
+import java.lang.invoke.MethodHandles;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.MonthDay;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.format.SignStyle;
+
+import org.hibernate.search.engine.logging.impl.Log;
+import org.hibernate.search.util.common.impl.TimeHelper;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public final class ParseUtils {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private ParseUtils() {
+		// Private constructor, do not use
+	}
+
+	public static String parseString(String value) {
+		return value;
+	}
+
+	public static Instant parseInstant(String value) {
```

Comment:
I'm not sure why you're using explicit formats everywhere while just calling `TheType.parse(String)` achieves exactly the same result, especially since the format is not likely to ever change.

Using `TheType.parse(String)` makes it easier to document we will only mention "use a string that can be parsed by `TheType.parse(String)`" and let users have a look at the javadoc of this method.

Reply:
I just noticed that you were displaying the formatter in the error message, so you probably wanted to actually use that formatter when parsing. That makes sense. Anyway, you already changed it, it will work just as well :)